gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapClient.java

```diff
@@ -109,15 +115,18 @@ public String getMemberOf(String user) {
     String filter = config.getLdapUserSearch().replace("${USER}", user);
 
     String[] attributes = new String[]{config.getLdapGroupMemberAttribute()};
-    List<UserRecord> list = ldapConnectionTemplate.search(config.getLdapUserBaseDn(),
+    List<Optional<UserRecord>> list = ldapConnectionTemplate.search(config.getLdapUserBaseDn(),
```

Comment:
nit: we should use more meaningful name than "list"

Reply:
How about `userRecords`?

Reply:
sounds good to me